### PR TITLE
Track 'iNZightVIT/Lite2' changes

### PR DESCRIPTION
**Pull request by tmelliott**
This branch forks off Lite at the point 'iNZightVIT/Lite2' was created and includes the file changes, with a few additions to hide sensitive information in ENV variables instead of hard-coding them - this removes the necessity for a separate, private repository.

The overall changes in this branch are
- suppression of small counts and the ability to see individual rows ("microdata")
- facilities for decoding an encrypted URL (here specifically for census at school, but should be generalisable)
- looks like there's also some random sampling facilities?

### CODE REVIEW

**Comment by chrk623** [comment]
@tmelliott Lite2 seems ready. Would you mind to do some testing incase I missed anything?

**Comment by tmelliott** [comment]
Thanks @chrk623 - there was a bug in the decryption (brackets in the wrong places). While I was there I removed the library calls and used prefixing instead - I know `library()` is used heavily but it seems silly to import libraries only used in one specific instance! Also, instead of manually hosting the hex2raw functions I've just imported the package (a bit of overhead but nothing too much in terms of what's already being installed..!).

**Comment by chrk623** [comment]
@tmelliott Sounds good.
I actually had trouble getting `wkb` to install with R 4.2.2 thus I did it manually. I'll need to go figure that out.

So I guess the PR is good to go?

File: panels/D10_UniteColumns/2_unite.columns.server.R [hunk 1]
```diff
@@ -27,8 +27,15 @@ observe({
       name = ifelse(input$name_unite_columns == "", "newcol", input$name_unite_columns)
       sep = input$sep_unite_columns
       temp = iNZightTools::unite(get.data.set(), name, col, sep)
+      
+      data.set = as.data.frame(temp)
```
Comment (tmelliott) [inline panels/D10_UniteColumns/2_unite.columns.server.R:31]: And again - there must be a better way than all this repetition?! A `values <- some_fun()`, perhaps? I'll let you play with that!

File: panels/D10_UniteColumns/2_unite.columns.server.R [hunk 2]
```diff
@@ -54,15 +61,22 @@ observe({
       code.save$variable = c(code.save$variable, list(c("\n", code)))
       ## save data
       updatePanel$datachanged = updatePanel$datachanged+1
-      values$data.set = temp
+      
+      values$data.set = as.data.frame(temp)
```
Comment (tmelliott) [inline panels/D10_UniteColumns/2_unite.columns.server.R:65]: LITE2

File: panels/D10_UniteColumns/2_unite.columns.server.R [hunk 3]
```diff
@@ -54,15 +61,22 @@ observe({
...
+      colnames(values$data.sample) = colnames(values$data.set)
+      
       code.save$name = code.save$dataname
       values$data.name = code.save$dataname
     }
   })
 })
 
 output$unitecolumns.table = renderDT({
-  get.data.set()
```
Comment (tmelliott) [inline panels/D10_UniteColumns/2_unite.columns.server.R:65]: if-else (again, `values$data.sample` is I think only in the LITE2 version, but I could be wrong ....).

File: panels/D11_MergeJoinDatasets/2_mergejoin.datasets.server.R [hunk 4]
```diff
@@ -121,8 +121,16 @@ observe({
       
       if(orig_type == new_type|orig_type == "character" & new_type == "factor"|orig_type == "factor" & new_type == "character") {
         temp.join = iNZightTools::joindata(data, newdata, left_col, right_col, join_method, left_name, right_name)
+        
+        data.set = as.data.frame(temp.join)
+        
+        sample.num = ifelse(nrow(data.set) > 2000, 500, round(nrow(data.set)/4))
```
Comment (tmelliott) [inline panels/D11_MergeJoinDatasets/2_mergejoin.datasets.server.R:127]: LITE2

File: panels/D11_MergeJoinDatasets/2_mergejoin.datasets.server.R [hunk 5]
```diff
@@ -196,7 +204,14 @@ observe({
         code.save$variable = c(code.save$variable, list(c("\n", code)))
         ## save data
         updatePanel$datachanged = updatePanel$datachanged+1
-        values$data.set = temp.join
+        
+        values$data.set = as.data.frame(temp.join)
```
Comment (tmelliott) [inline panels/D11_MergeJoinDatasets/2_mergejoin.datasets.server.R:208]: LITE2

File: panels/D11_MergeJoinDatasets/2_mergejoin.datasets.server.R [hunk 6]
```diff
@@ -275,8 +290,16 @@ observe({
         }
       }
       temp.append = iNZightTools::appendrows(data, newdata, date)
+      
+      data.set = as.data.frame(temp.append)
+      
+      sample.num = ifelse(nrow(data.set) > 2000, 500, round(nrow(data.set)/4))
```
Comment (tmelliott) [inline panels/D11_MergeJoinDatasets/2_mergejoin.datasets.server.R:296]: LITE2

File: panels/D12_FrequencyTables/1_frequency.tables.ui.R [hunk 7]
```diff
@@ -28,7 +28,9 @@ frequency.tables.panel = function(data.set){
   }else{
     sidebarLayout(
       sidebarPanel(frequency.tables.sidebar()),
-      mainPanel(DTOutput("ft.expand.data.table"))
```
Comment (tmelliott) [inline panels/D12_FrequencyTables/1_frequency.tables.ui.R:31]: LITE2 - if else maybe?

**Review by tmelliott — COMMENTED** [review]
Still some work to do, but it's "working" now! I believe there are (many?) places where there were no merge conflicts, but the added code should be conditional based on the `LITE2` variable. I've started going though and pointing out some of the places where this is the case, but there are lots of them ... in one comment I suggest a wrapper fun to replace a lot of the duplicated `values` code, because that kind of repetition is nasty... 

There are many more places but I haven't scrolled that far (btw, working off https://github.com/iNZightVIT/Lite/pull/293/files) but a lot seem to fit a pattern. Once you've had a go at a few, I'll take a look.

Any questions, let me know!

**Comment by tmelliott** [comment]
> I actually had trouble getting `wkb` to install with R 4.2.2 thus I did it manually. I'll need to go figure that out.
What OS are you using? The issue was probably with installing the `sp` dependency (which is needed for the Maps stuff anyway so in theory Lite already imports it).

**Comment by chrk623** [comment]
> > I actually had trouble getting `wkb` to install with R 4.2.2 thus I did it manually. I'll need to go figure that out.
> 
> What OS are you using? The issue was probably with installing the `sp` dependency (which is needed for the Maps stuff anyway so in theory Lite already imports it).

Ubuntu 22.04 but somehow i just successfully installed it... probably had to install something and I didnt restart the R instance...

Totally forgot about the other panels. Seems like for some places, `if-else` is not strictly required, for example [this](https://github.com/iNZightVIT/Lite/pull/293/files/110845e73a4ed7c6b2b71048de19ed68de8d8d4e#r1285279883), unless we want a smaller overhead?

**Comment by tmelliott** [comment]
I think that was supposed to be an `if LITE2`. Do what you think is appropriate - the main thing is making all those sampling and additional bits of UI display only in the case of LITE2, but otherwise it should be exactly the same as the normal Lite.

**Comment by chrk623** [comment]
@tmelliott Should be good, though I need to do some more testing as there are changes to too many panels. Feel free to take a look if you have time.

File: functions.R [hunk 8]
```diff
@@ -2,6 +2,21 @@
 # To be removed when the iNZight tools package is working##
 ##########################################################
 
+# LITE2
+sample_if_lite2 = function(rvalues, d, new_sample = TRUE) {
+  if(LITE2) {
```
Comment (tmelliott) [inline functions.R]: Good coding trick in these situations:
```
if (!LITE2) return(rvalues)
...rest of code...
```

**Comment by tmelliott** [comment]
Great @chrk623 - looking much better already. I probably wont have a chance to do any testing this week, but nudge me if you need anything!

**Comment by chrk623** [comment]
Seems good now, merge first? or deal with the urls and access keys? @tmelliott

**Comment by tmelliott** [comment]
If you think you've captured all the differences in if-else blocks, then yep - go ahead and merge (use the drop-down to pick 'squash and merge' to keep the git history simpler).

**Pull request merged**